### PR TITLE
security: fix resource leaks from staticmajor + add cosmos/gosec & staticmajor

### PR DESCRIPTION
**Pull request by odeke-em**
This change fixes resource leaks identified by a static analyzer from Orijtech Inc called "staticmajor".

While here, also added cosmos/gosec scanning which will help with running security rules and take advantage of Github's vulnerability scanners on runs.

### CODE REVIEW

**Comment by mmulji-ic** [comment]
Thanks @odeke-em looks like we have issues we need to fix!

**Comment by odeke-em** [comment]
Cheers @mmulji-ic! Don't worry though, the gosec issues can be progressively fixed and will just be filed in a section on GitHub security and eventually can look at them. A bunch of them might even be false positives. The merge of this PR shouldn't hold y'all back :-)

**Review by mmulji-ic — APPROVED** [review]

**Comment by odeke-em** [comment]
Howdy my good folks, please help with a merge.. This code has lingered for more than a month :-)

File: .github/workflows/gosec.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Checkout Source
+        uses: actions/checkout@v3
+
+      - name: Run Gosec Security Scanner
+        uses: cosmos/gosec@master
+        with:
+          # Let the report trigger a failure with the Github Security scanner features.
+          args: "-no-fail -fmt sarif -out results.sarif ./..."
+
+      - name: Upload SARIF file
```
Comment (yaruwangway) [inline .github/workflows/gosec.yml:34]: Hi, this will add `results.sarif` to gaia repo ? is this a necessity ?

Comment (Pantani) [reply .github/workflows/gosec.yml:34]: I think we need the Sarif file to show the results on GitHub. But I'm not 100% sure

https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning

**Comment by okwme** [comment]
priority is to confirm whether this new check has exposed anything that needs to be fixed before Rho is released.
If nothing needs to be changed, let's wait until after Rho as changes in Lambda may make many of these moot to begin with.
If there are changes needed, let's make those changes first and then merge this after.
@Pantani will run through the list before Rho release.

**Comment by Pantani** [comment]
I think we can exclude some rules like `G107 (CWE-88)` and `G701 (CWE-)`

**Comment by Pantani** [comment]
@odeke-em, can you please update your branch?

**Pull request closed without merging**